project/dexter-eclipse/src/java/com/samsung/sec/dexter/eclipse/action/OpenCodeMetricsDelegate.java

```diff
@@ -0,0 +1,178 @@
+
+public class OpenCodeMetricsDelegate implements IObjectActionDelegate {
+
+	ISelection selection;
+	String modulePath = "";
+	String fileName = "";
+	
+	@Override
+	public void run(IAction action) {
+		if (selection instanceof StructuredSelection) {
+
+			final StructuredSelection sel = (StructuredSelection) this.selection;
+
+			@SuppressWarnings("unchecked")
+			Iterator<Object> iter = sel.iterator();
+			while (iter.hasNext()) {
+				final Object object = iter.next();
+				if (object instanceof IResource) {
+					final IResource resource = (IResource) object;
+					codeMetricsResource(resource);
+				}
+			}
+
+		}
+	}
+
+	private void codeMetricsResource(final IResource resource) {
+		if (resource instanceof IFile) {
+			final IFile targetFile = (IFile) resource;
+			if (targetFile.getName().endsWith(".java")) {
+
+				setModulePath(getModulePath(targetFile));
+				setFileName(targetFile.getName());
+				StringBuilder url = new StringBuilder();
+				try {
+					IViewPart view = EclipseUtil.findView(CodeMetricsView.ID);
+					final CodeMetricsView codeMetricsView = (CodeMetricsView) view;
+
+					
+					url.append("http://").append(DexterClient.getInstance().getServerHost()).append(":") //$NON-NLS-1$ //$NON-NLS-2$
+							.append(DexterClient.getInstance().getServerPort())
+							.append(DexterConfig.CODE_METRICS_BASE)
+							//$NON-NLS-1$
+							.append("?").append(DexterConfig.CODE_METRICS_FILE_NAME).append("=").append(getFileName())//$NON-NLS-1$
+							.append("&").append(DexterConfig.CODE_METRICS_MODULE_PATH).append("=").append(getModulePath());//$NON-NLS-1$
+					
+					DexterEclipseActivator.LOG.info(url.toString());
+					codeMetricsView.setUrl(url.toString());
+
+					EclipseUtil.showView(CodeMetricsView.ID);
+				} catch (DexterRuntimeException e) {
+					DexterEclipseActivator.LOG.error("Cannot open the Code Metrics Description View");
+					DexterEclipseActivator.LOG.error(e.getMessage(), e);
+				}
+			}
+		}
+	}
+
+	private void setModulePath(String modulePath) {
+		if (modulePath.startsWith("/")) {
+			modulePath = modulePath.substring(1, modulePath.length());
+		}
+		this.modulePath = modulePath;
+	}
+
+	private void setFileName(String fileName) {
+		this.fileName = fileName;
+	}
+
+	final private String getModulePath() {
+		return this.modulePath;
+	}
+
+	final private String getFileName() {
+		return this.fileName;
+	}
+
+	public static IJavaProject getJavaProject(final IFile file) {
+		final IProject project = file.getProject();
+		if (project == null) {
+			throw new DexterRuntimeException("Project is null");
+		}
+
+		IJavaProject javaProject;
+		try {
+			if (project.hasNature(JavaCore.NATURE_ID)) {
+				javaProject = JavaCore.create(project);
+			} else {
+				throw new DexterRuntimeException("this is not java project");
+			}
+		} catch (CoreException e1) {
+			throw new DexterRuntimeException(e1.getMessage(), e1);
+		}
+
+		return javaProject;
+	}
+
+	public String getModulePath(IFile file) {
```

Comment:
Please move the method into EclipseUtil.java. if there is same method, please use that one.